elyra/util/kubernetes.py

```diff
@@ -37,3 +38,16 @@ def is_valid_kubernetes_resource_name(name: str) -> bool:
         elif char not in ["-", "."]:
             return False
     return True
+
+
+def is_valid_kubernetes_key_name(name: str) -> bool:
+    """
+    Returns a truthy value indicating whether name meets the kubernetes
+    naming constraints, as outlined in the link below.
+
+    https://kubernetes.io/docs/concepts/configuration/secret/#restriction-names-data
```

Comment:
It appears secrets and volumes are "validated" just prior to platform submission.  Should these kinds of resources (names and keys) be validated during pipeline validation and converted to their "usable forms" then?  If so, we should probably ensure length is `< MiB`.